pkg/utils/maputils.go

```diff
@@ -69,21 +69,20 @@ func CreateRecord(columnNames []string, record []string) (map[string]interface{}
 // SetDifference returns the added and removed elements between two sets.
 func SetDifference[T1, T2 any](newSet map[string]T2, oldSet map[string]T1) ([]string, []string, bool) {
 	var added, removed []string
-	change := false
+	newEntries := false
 
 	for key := range oldSet {
 		if _, exists := newSet[key]; !exists {
 			removed = append(removed, key)
-			change = true
 		}
 	}
 
 	for key := range newSet {
 		added = append(added, key)
 		if _, exists := oldSet[key]; !exists {
-			change = true
+			newEntries = true
 		}
 	}
 
-	return added, removed, change
+	return added, removed, newEntries
```

Comment:
nit: for a boolean, I think `newEntries` is a little unclear because (without type information) it could either mean "a list of keys that were added" or "the sets are different". Maybe `hasChange` or `setsDiffer` would be slightly better.

Also, is this `newEntries` variable actually useful? It seems like the caller could get the same info from `len(added) > 0`, right?

Reply:
No len(added) will always be greater than 0, if PQIDs are there. Added records will actually be old records that are present in the newSet + newly added records in the newSet.

Reply:
Oh, is that actually what you want to return (I'm talking about the `added` variable)? If it is, then you need to change some documentation to clarify that:
- `added` should be renamed to something indicating that it's all of the keys in the new set, not just the ones added to the old set to make the new set
- In you function comment: `// SetDifference returns the added and removed elements between two sets.` you should also make that clarification

Reply:
Upon clarification, we will only log changes that means we will only log newly added entries in promoted and removed entries in demoted. Have removed this newEntries/change flag and updated the function and tests for the same.